### ISSUE TITLE
Working around bugs in the pb program

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/policybrain-builder#70]
For the most part, the `pb` program, which was written by @jbcrail, is a big improvement in our ability to build and upload conda packages for OSPC projects.  Thanks for the helpful tool, @jbcrail.

At this writing, there seem to be a couple of bugs that prevent `pb` from working as advertised.

The purpose of this post is to give an example of a bash script that works around these problems.
The example script is for creating taxcalc packages, but the `pb release` command can be edited to build and upload other OSPC packages.
```
#!/bin/bash
# Script to build and upload taxcalc package for specified release using
# policybrain-builder CLI, pb, with a single parameter, TAG (eg, 0.11.0).
# NOTE: --clean option doen't work, so never use -w or --workdir options.
# NOTE: --python option doesn't work either, so use OSPC_PYTHONS instead.

echo "STARTING : `date`"

rm -rf /tmp/policybrain-builder 

export OSPC_PYTHONS="2.7 3.6"
TOKEN=<secret-token>
pb release --token $TOKEN --label main taxcalc=$1 2>&1 | awk '$1~/\[taxcalc\]/'
unset OSPC_PYTHONS

conda build purge
find ~/anaconda/conda-bld -name "*tar.bz2" -exec rm -f {} \;

echo "FINISHED : `date`"
exit 0
```
Creating a temporary test release 0.1.0 and running the script produces ten conda packages (uploaded to `https://anaconda.org/ospc/taxcalc` and the following output:
```
./taxcalc-package-cli.sh 0.1.0
STARTING : Fri Oct 13 16:38:12 EDT 2017
[taxcalc] removing
[taxcalc] cloning
[taxcalc] fetching
[taxcalc] checking out '0.1.0'
[taxcalc] archiving
[taxcalc] building 3.6
[taxcalc] converting to linux-32
[taxcalc] converting to linux-64
[taxcalc] converting to win-32
[taxcalc] converting to win-64
[taxcalc] building 2.7
[taxcalc] converting to linux-32
[taxcalc] converting to linux-64
[taxcalc] converting to win-32
[taxcalc] converting to win-64
[taxcalc] uploading osx-64 packages
[taxcalc] uploading linux-32 packages
[taxcalc] uploading linux-64 packages
[taxcalc] uploading win-32 packages
[taxcalc] uploading win-64 packages
FINISHED : Fri Oct 13 16:55:06 EDT 2017
```

@MattHJensen @jdebacker @rickecon @jbcrail

**Comment by jbcrail** [comment]
@martinholmer I fixed #67 (`--clean` issue) and #69 (`--python` issue).

Let me know if there are other issues.

**Comment by martinholmer** [comment]
@jbcrail said:
> I fixed #67 (--clean issue) and #69 (--python issue).
>
> Let me know if there are other issues.

Great!  Thank you.  I'll test these changes later today.

One more thing: can you close issue #45?  Seems like #45 is obsolete.

**Comment by martinholmer** [comment]
@jbcrail, thanks for fixing the bugs mentioned in issue #70.
